Tests/swift-sdk-swift-tests/InAppHelperTests.swift

```diff
@@ -304,45 +304,151 @@ class InAppHelperTests: XCTestCase {
         TestUtils.validateMatch(keyPath: KeyPath("myPayload.var1"), value: "val1", inDictionary: dict, message: "Expected to find val1")
     }
     
-    // This tests payload as we expect to get when backend is fixed
-    func testInAppPayloadParsing() {
-        let customPayload1 = """
+    // Remove this test when backend is fixed
+    // This test assumes that certain parts of payload
+    // are in 'customPayload' element instead of the right places.
+    func testInAppPayloadParsingWithPreprocessing() {
+        let customPayloadStr1 = """
         {
-            "contentType" : "html",
-            "inAppType" : "default",
-            "channelName" : "channel1"
+            "messageId": "overridden",
+            "var1" : "value1",
+            "obj1" : {
+                "something" : true,
+                "nothing" : "is nothing"
+            },
         }
         """
-        let customPayload2 = """
+        var customPayload1 = customPayloadStr1.toJsonDict()
+        customPayload1["inAppType"] = "default"
+        customPayload1["contentType"] = "html"
+        
+        let customPayloadStr2 = """
         {
-            "contentType" : "html",
-            "inAppType" : "inBox",
-            "channelName" : "channel2",
             "promoteToContent" : {
                 "title" : "title",
                 "subTitle" : "subtitle",
                 "imageUrl" : "http://somewhere.com/something.jpg"
             }
         }
         """
+        var customPayload2 = customPayloadStr2.toJsonDict()
+        customPayload2["inAppType"] = "inBox"
```

Comment:
I think since the feature itself is `Inbox`, the internal name also should be `inbox` (all lowercase, without capitalization in the middle)